sycl/include/CL/sycl/kernel_bundle.hpp

```diff
@@ -572,16 +555,25 @@ compile(const kernel_bundle<bundle_state::input> &InputBundle,
 namespace detail {
 std::vector<sycl::device> find_device_intersection(
     const std::vector<kernel_bundle<bundle_state::object>> &ObjectBundles);
+
+__SYCL_EXPORT std::shared_ptr<detail::kernel_bundle_impl>
+link_impl(const std::vector<kernel_bundle<bundle_state::object>> &ObjectBundles,
+          const std::vector<device> &Devs, const property_list &PropList);
 }
 
 /// \returns a new kernel_bundle which contains the device images from the
 /// ObjectBundles that are translated into one or more new device images of
 /// state bundle_state::executable The new bundle represents all of the kernels
 /// in ObjectBundles that are compatible with at least one of the devices in
 /// Devs.
-kernel_bundle<bundle_state::executable>
+inline kernel_bundle<bundle_state::executable>
```

Comment:
Just curious, why are `compile()`, `link()`, and `build()` declared `inline` while none of the other functions in this header use `inline`?

Reply:
The non-template functions which have definition in the header itself are marked with `inline` to avoid multiple definitions error.